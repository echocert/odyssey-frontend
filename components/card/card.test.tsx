import React from 'react';
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react';
import Card from './card';

describe('Card Component', () => {
    it("Verify the snapshot of questionnaire card is matched", () => {
        const wrap = create(<Card cardTitle={''} cardUnit={''} cardType={''} cardAddress={''} />).toJSON();
        expect(wrap).toMatchSnapshot();
    });
    it('renders a card with title, address, and units', () => {
        const cardProps = {
            cardTitle: 'Sample Title',
            cardUnit: 42,
            cardType: 'property',
            cardAddress: '123 Main St',
        };

        render(<Card {...cardProps} />);
        const cardTitleText = screen.getByTestId('card-title');
        expect(cardProps.cardTitle).toEqual(cardTitleText.innerHTML);
        const cardAddressText = screen.getByTestId('card-address');
        expect(cardProps.cardAddress).toEqual(cardAddressText.innerHTML);



        // const cardTypeElement = screen.queryByTestId('card-type');
        // expect(cardTypeElement).toBeInTheDocument();

    });

    it('calls editClick when the edit icon is clicked', () => {
        const editClickMock = jest.fn();
        const cardProps = {
            cardTitle: 'Sample Title',
            cardUnit: 42,
            cardType: 'property',
            cardAddress: '123 Main St',
            editClick: editClickMock,
        };

        render(<Card {...cardProps} />);

        // const editIcon = screen.getByTestId('edit-icon');

        // fireEvent.click(editIcon);


        // expect(editClickMock).toHaveBeenCalled();
    });
});
