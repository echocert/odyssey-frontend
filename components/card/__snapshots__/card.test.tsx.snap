// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Verify the snapshot of questionnaire card is matched 1`] = `
<div
  className="card"
>
  <div />
  <div>
    <div
      className="card__title"
    >
      <label
        className="label-l4 label-header"
        data-testid="card-title"
        style={{}}
      />
    </div>
    <div
      className="card__address"
    >
      <label
        className="label-subheader-l3 label-l3 label-subheader"
        data-testid="card-address"
        style={{}}
      />
    </div>
    <div
      className="card__details"
    >
      <div
        className="card__units"
      >
        <div>
          <span>
            <label
              className="label-subheader-l3 label-l3 label-subheader"
              data-testid="card-unit"
              style={{}}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
